frontend/src/components/phones/dashboard/Dashboard.tsx

```diff
@@ -209,112 +208,27 @@ export const PhoneDashboard = () => {
-          >
-            {data.last_inbound_type === "text" && (
-              <ForwardedTextIcon
-                alt="Last received a text"
-                className={styles["forwarded-type-icon"]}
-                width={20}
-                height={12}
-              />
-            )}
-            {data.last_inbound_type === "call" && (
-              <ForwardedCallIcon
-                alt="Last received a call"
-                className={styles["forwarded-type-icon"]}
-                width={20}
-                height={15}
-              />
-            )}
-            <Moment calendar={calendarStrings}>{data.last_inbound_date}</Moment>
-          </span>
-          <span className={styles["sender-controls"]}>
-            <button
-              onClick={() =>
-                inboundContactData.setForwardingState(!data.blocked, data.id)
-              }
-              className={styles["block-btn"]}
-            >
-              {data.blocked ? "Unblock" : "Block"}
-            </button>
-          </span>
-        </li>
-      );
-    });
-
-  const callerSMSSendersPanel = (
-    <div id="secondary-panel" className={styles["dashboard-card"]}>
-      <div className={styles["dashboard-card-caller-sms-senders-header"]}>
-        <span>
-          <button
-            type="button"
-            onClick={toggleSendersPanel}
-            className={styles["caller-sms-logs-back-btn"]}
-          >
-            <ChevronLeftIcon
-              alt="Back to Primary Dashboard"
-              className={styles["nav-icon"]}
-              width={20}
-              height={20}
-            />
-          </button>
-        </span>
-        <span className={styles["caller-sms-logs-title"]}>
-          {l10n.getString("phone-dashboard-senders-header")}
-        </span>
-        <span></span>
-      </div>
-      <ul className={styles["caller-sms-senders-table"]}>
-        <li className={styles["greyed-contact"]}>
-          <span>
-            {l10n.getString("phone-dashboard-sender-table-title-sender")}
-          </span>
-          <span>
-            {l10n.getString("phone-dashboard-sender-table-title-activity")}
-          </span>
-          <span>
-            {l10n.getString("phone-dashboard-sender-table-title-action")}
-          </span>
-        </li>
-        {inboundContactArray}
-      </ul>
-    </div>
-  );
+  function setSendersPanelView() {
+    if (profileData.data?.[0].store_phone_log === false) {
+      return "disabled";
+    }
+    if (inboundArray && inboundArray.length === 0) {
+      return "empty";
+    }
+    return "primary";
+  }
 
   return (
     <main className={styles["main-phone-wrapper"]}>
-      {showingPrimaryDashboard ? (
+      {showingPrimaryDashboard && inboundContactData !== undefined ? (
         // Primary Panel
         <>{primaryPanel}</>
       ) : (
         // Caller and SMS Senders Panel
-        <>{callerSMSSendersPanel}</>
+        <SendersPanelView
+          type={setSendersPanelView()}
```

Comment:
nit (non-blocking): `setSendersPanelView` seems like this function will set some kind of state. A more appropriate function name would be `getSendersPanelType()`